### ISSUE TITLE
(Bug report) 

### ISSUE THREAD

**Issue by HyperCriSiS** [zadam/trilium#4192]
### Trilium Version

0.60.4

### What operating system are you using?

Windows

### What is your setup?

Local + server sync

### Operating System Version

Windows 11

### Description

Hi,

just switched from Joplin to Trilium. Went very well without issues!
I just try to setup the desktop application in full portable mode, but this seems not to work as expected.
How can I start it without having the cmd window open every time?
Also it still stores files in the roaming folder.

Best regards
Hyper

**Comment by Nriver** [comment]
Just double click `trilium-portable.bat`, the data should be stored under the same folder of the program.

**Comment by HyperCriSiS** [comment]
Yes but not everything, it still uses the roaming folder for files during runtime.
Also it is annoying that the terminal stays open.

**Comment by HyperCriSiS** [comment]
As I just see, after a Windows reboot, the cmd closes after Trilium was started.
But when it is closed and opened again with the portable batch, it stays open and shows logs.

**Comment by zadam** [comment]
Check what the `trilium-portable.bat` does - it sets a TRILIUM\_DATA\_DIR environment variable. If you set this into the global environment variables (IIRC somewhere in the System settings, then you don't need to use the .bat and there won't be any cmd window.

Regarding the files in the roaming folder (IIRC called something like "Trilium Notes" as opposed to "trilium-data") - those are created by electron itself and I can't really control that (AFAIK). But they are mostly just caches, spelling dictionary downloads etc. There's no "valuable" data stored there and will be recreated as needed by the electron.

**Comment by HyperCriSiS** [comment]
> Check what the `trilium-portable.bat` does - it sets a TRILIUM_DATA_DIR environment variable. If you set this into the global environment variables (IIRC somewhere in the System settings, then you don't need to use the .bat and there won't be any cmd window.
> 
> Regarding the files in the roaming folder (IIRC called something like "Trilium Notes" as opposed to "trilium-data") - those are created by electron itself and I can't really control that (AFAIK). But they are mostly just caches, spelling dictionary downloads etc. There's no "valuable" data stored there and will be recreated as needed by the electron.

Thanks, I'll try that.
Why don't you just add a parameter for the trilium.exe? for example starting with trilium.exe /p
That would be more convenient.

**Comment by u1735067** [comment]
I never noticed the terminal/console window until today (and not on every computer), but another solution is https://www.electronjs.org/docs/latest/api/environment-variables#electron_no_attach_console-windows: set the `ELECTRON_NO_ATTACH_CONSOLE` environment variable (to `1` for example, see https://github.com/electron/electron/blob/v27.0.4/shell/app/electron_main_win.cc#L155)

The portable script can be written this way:
```batch
SET DIR=%~dp0
SET TRILIUM_DATA_DIR=%DIR%\trilium-data
SET ELECTRON_NO_ATTACH_CONSOLE=1
cd %DIR%
start trilium.exe
```

For the Electron `userData` files, it seems possible to control that:
* https://stackoverflow.com/questions/48587035/electron-how-to-set-a-custom-directory-for-user-data-user-data-dir
* https://www.electronjs.org/docs/latest/api/app#appsetpathname-path
* `app.setPath('userData', "path/to/new/directory");`

Ideally it should be configurable (to a path different than `TRILIUM_DATA_DIR`)

**Comment by KommissarKong** [comment]
Sorry to comment on a zombie thread but the question only fits here.
The last comment was about moving the roaming profile to a custom path but am I right in my understanding that this still isn't implemented?
I use Trilium in my company and our roaming profile is about 100MB, and the trilium roaming profile sucks it up too fast.
I'd love to spread the application to colleagues but this way I can't.

Am I wrong in my understanding or will there be any updatesregarding this?

**Comment by eliandoran** [comment]
@KommissarKong , interesting limitation.

Unfortunately this has not been addressed yet. This is a global issue with Electron-based applications (and not only). For example, my `AppData/Roaming` has 38 different applications, totaling 13.4 GB.

You can change the data directory, though, although I suppose you already did that since this thread is about portability.

We'll look into it for the next release.

**Comment by KommissarKong** [comment]
@eliandoran Very much appreciated!
I'll humbly delete my roaming profile every day till the fix will hopefully arrive. 
(づ￣ 3￣)づ
